ansible_collections/arista/avd/roles/eos_validate_state/templates/validate_state_report-csv.j2

```diff
@@ -1,230 +1,28 @@
-{%             if result.skipped is not defined %}
-{%                 set test_id.value = test_id.value + 1 %}
-{{ test_id.value }},{{ node }},Routing Table,Remote VTEP address,{{ result.vtep_reachability_test.vtep_ip }},{% if result.failed == false %}PASS{% else %}FAIL,{{ result.msg }}{% endif %}
-
-{%             endif %}
-{%         endfor %}
-{%     endif %}
-{% endfor %}
-{# Routing table check: loopback0 Results #}
-{% for node in groups[fabric_name] | arista.avd.natural_sort %}
-{%     if hostvars[node].routing_table_loopback0_results is defined and hostvars[node].routing_table_loopback0_results.results is defined %}
-{%         for result in hostvars[node].routing_table_loopback0_results.results %}
-{%             if result.skipped is not defined %}
-{%                 set test_id.value = test_id.value + 1 %}
-{{ test_id.value }},{{ node }},Routing Table,Remote Lo0 address,{{ result.routing_table_loopback0_test.loopback0_address }},{% if result.failed == false %}PASS{% else %}FAIL,{{ result.msg }}{% endif %}
-
-{%             endif %}
-{%         endfor %}
-{%     endif %}
-{% endfor %}
-{# loopback0 reachability Results #}
-{% for node in groups[fabric_name] | arista.avd.natural_sort %}
-{%     if hostvars[node].loopback0_reachability_results is defined and hostvars[node].loopback0_reachability_results.results is defined %}
-{%         for result in hostvars[node].loopback0_reachability_results.results %}
-{%             if result.skipped is not defined %}
-{%                 set test_id.value = test_id.value + 1 %}
-{{ test_id.value }},{{ node }},Loopback0 Reachability,Loopback0 Reachability,Source: {{ node }} - {{ hostvars[node].loopback_interfaces.Loopback0.ip_address | ansible.netcommon.ipaddr('address') }} Destination: {{ result.loopback0_reachability_test.loopback0_address }},{% if result.failed == false %}PASS{% else %}FAIL,{{ result.msg }}{% endif %}
-
-{%             endif %}
-{%         endfor %}
-{%     endif %}
-{% endfor %}
-{# Inb management to loopback 0 reachability Results #}
-{% for node in groups[fabric_name] | arista.avd.natural_sort %}
-{%     if hostvars[node].inb_mgmt_loopback0_reachability_results.results is arista.avd.defined %}
-{%         for result in hostvars[node].inb_mgmt_loopback0_reachability_results.results %}
-{%             if result.skipped is not defined %}
-{%                 set test_id.value = test_id.value + 1 %}
-{{ test_id.value }},{{ node }},Loopback0 Reachability,Inband Mgmt Reachability,Source: {{ node }} - {{ result.inb_mgmt_loopback0_reachability_test.invocation.module_args.commands[0].split(" ") | last }} Destination: {{ result.inb_mgmt_loopback0_reachability_test.loopback0_address }},{% if result.failed == false %}PASS{% else %}FAIL,{{ result.msg }}{% endif %}
-
-{%             endif %}
-{%         endfor %}
+{% for report_entry in eos_validate_state_report | arista.avd.default([]) %}
+{%     if only_failed_tests is arista.avd.defined(false) %}
```

Comment:
the logic should be reversed. We should only run the special case if the value is `true`. Otherwise any value (even an empty string) would trigger the "only_failed_tests" scenario.
```suggestion
{%     if only_failed_tests is not arista.avd.defined(true) %}
```